### ISSUE TITLE
Bug: Beds in 2 high rooms are not sleep-able (but can be claimed)

### ISSUE THREAD

**Issue by tomerorian** [TangoTek/TekTopia-Community#43]
**Describe the bug**
A bed with a roof 2 blocks above its floor (so 1 block gap) is valid and claimable, but the villager will not be able to sleep in it.

**To Reproduce**
1. Build a house 2 block high with a bed
2. Villager will stay near bed all night, but not sleep

**Expected behavior**
Either:
- The villager should be able to sleep in the bed
- It should not be valid

**Screenshots**
![image](https://user-images.githubusercontent.com/2083414/58381866-9c5dea80-7fcb-11e9-9a8e-ee9b8fa15783.png)

**Comment by Geex204** [comment]
Can confirm. Even two and a half blocks are not enough.
![2019-06-03_12 37 46](https://user-images.githubusercontent.com/51030223/58796060-7accb680-85fc-11e9-9860-352d9cbcdcb8.png)

**Comment by JohnyDL** [comment]
Yeah this is cause the villager's hit box doesn't change when they're asleep. They aren't lying down they're standing on their bed with an animation and beds are 0.575 high iirc... a trap door should work though
![2019-05-30_19 13 08](https://user-images.githubusercontent.com/5202568/58798353-5243bd00-85fa-11e9-9dae-61bb887eafa0.png)

**Comment by Minetrain2002** [comment]
villagers now sleep in the beds but get stuck when they wake up
![2019-07-24_17 35 51](https://user-images.githubusercontent.com/52817504/62125119-86441180-b30f-11e9-9f48-26b65632d75a.png)
